hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/MappingCatalog.java

```diff
@@ -71,7 +75,9 @@ public void createMapping(Mapping mapping, boolean replace, boolean ifNotExists)
         if (ifNotExists) {
             storage.putIfAbsent(name, resolved);
         } else if (replace) {
-            storage.put(name, resolved);
+            if (storage.put(name, resolved) != null) {
```

Comment:
A new mapping can invalidate cached plans even if it doesn't overwrite an older mapping - it can shadow an implicit mapping.